File: Java-PerfKata/src/test/java/perfkata/parrot/a20171110a/AfricanParrot.java
package perfkata.parrot.a20171110a;

public class AfricanParrot extends Parrot {

    private final int numberOfCoconuts;

    public AfricanParrot(int numberOfCoconuts) {
        this.numberOfCoconuts = numberOfCoconuts;
    }

    @Override
    public double getSpeed() {
        return Math.max(0, getBaseSpeed() - slowDownDueToCoconuts());
    }

    private double slowDownDueToCoconuts() {
        return getLoadFactor() * numberOfCoconuts;
    }

    private double getLoadFactor() {
        return 9.0;
    }

}

File: TypeScript/jest/src/parrots/AfricanParrot.ts
import { IParrot } from './IParrot';
import { PARROT_BASE_SPEED } from './ParrotBaseSpeed';

const LOAD_FACTOR = 9;

export class AfricanParrot implements IParrot {
    constructor(private numberOfCoconuts: number) {
    }

    public getSpeed(): number {
        const weightSpeedPenalty = LOAD_FACTOR * this.numberOfCoconuts;
        return Math.max(0, PARROT_BASE_SPEED - weightSpeedPenalty);
    }
}
